posthog/api/person.py

```diff
@@ -613,11 +452,207 @@ def update(self, request, *args, **kwargs):
+                {"message": "Missing parameter", "detail": "Must include lifecycle type"}, status=400
+            )
+
+        people = self.lifecycle_class().get_people(
+            target_date=target_date_parsed, filter=filter, team=team, lifecycle_type=lifecycle_type,
+        )
+        next_url = paginated_result(people, request, filter.offset, filter.limit)
+        return response.Response({"results": [{"people": people, "count": len(people)}], "next": next_url})
+
+    @action(methods=["GET"], detail=False)
+    def retention(self, request: request.Request) -> response.Response:
+        display = request.GET.get("display", None)
+        team = cast(User, request.user).team
+        if not team:
+            return response.Response(
+                {"message": "Could not retrieve team", "detail": "Could not validate team associated with user"},
+                status=400,
+            )
+        filter = RetentionFilter(request=request, team=team)
+        filter = prepare_actor_query_filter(filter)
+        base_uri = request.build_absolute_uri("/")
+
+        if display == TRENDS_TABLE:
+            people = self.retention_class(base_uri=base_uri).actors_in_period(filter, team)
+        else:
+            people = self.retention_class(base_uri=base_uri).actors(filter, team)
+
+        next_url = paginated_result(people, request, filter.offset, filter.limit)
+
+        return response.Response({"result": people, "next": next_url})
+
+    @action(methods=["GET"], detail=False)
+    def stickiness(self, request: request.Request) -> response.Response:
+        team = cast(User, request.user).team
+        if not team:
+            return response.Response(
+                {"message": "Could not retrieve team", "detail": "Could not validate team associated with user"},
+                status=400,
+            )
+        filter = StickinessFilter(request=request, team=team, get_earliest_timestamp=get_earliest_timestamp)
+        filter = prepare_actor_query_filter(filter)
+
+        target_entity = get_target_entity(filter)
+
+        people = self.stickiness_class().people(target_entity, filter, team, request)
+        next_url = paginated_result(people, request, filter.offset, filter.limit)
+        return response.Response({"results": [{"people": people, "count": len(people)}], "next": next_url})
+
 
 def paginated_result(
-    entites: Union[List[Dict[str, Any]], ReturnDict], request: request.Request, offset: int = 0,
+    entites: Union[List[Any], ReturnDict], request: request.Request, offset: int = 0, limit: int = DEFAULT_PAGE_LIMIT,
 ) -> Optional[str]:
-    return format_paginated_url(request, offset, 100) if len(entites) > 99 else None
+    return format_paginated_url(request, offset, limit) if len(entites) >= limit else None
+
+
+T = TypeVar("T", Filter, PathFilter, RetentionFilter, StickinessFilter)
+
+
+def prepare_actor_query_filter(filter: T) -> T:
+    if not filter.limit or filter.limit > MAX_PAGE_LIMIT:
```

Comment:
reasons behind the MAX_PAGE_LIMIT?

Reply:
Seemed like a good idea 😅 More the question is "do we want an unrestricted limit to downloading of people".

I'm not even sure at what number this becomes a problem. As you mention it, it probably needs to be higher for CSV exports....

Reply:
Removed it as it might break things. Can always add it later